src/main/java/frc/trigon/robot/subsystems/arm/ArmConstants.java

```diff
@@ -0,0 +1,100 @@
+    private static final SensorDirectionValue ANGLE_MOTOR_DIRECTION = SensorDirectionValue.Clockwise_Positive;
+    private static final AbsoluteSensorRangeValue ANGLE_MOTOR_RANGE = AbsoluteSensorRangeValue.Signed_PlusMinusHalf;
+    private static final int
+            P = 0,
+            I = 0,
+            D = 0;
+    private static final PIDController PID_CONTROLLER_ANGLE = new PIDController(P, I, D);
+    private static final PIDController PID_CONTROLLER_ELEVATOR = new PIDController(P, I, D);
+
+    public enum ArmState {
+        DEFAULT(Rotation2d.fromDegrees(0), 0), TAKE_HIGH_CONE(Rotation2d.fromDegrees(60), 10), TAKE_GROUND_CONE(Rotation2d.fromDegrees(0), 0), PLACE_LOW_CONE(Rotation2d.fromDegrees(20), 3), PLACE_MEDIUM_CONE(Rotation2d.fromDegrees(50), 7), PLACE_HIGH_CONE(Rotation2d.fromDegrees(70), 10);
+
+        final Rotation2d angle;
+        final double elevatorPosition;
+
+        ArmState(Rotation2d angle, double elevatorPosition) {
+            this.angle = angle;
+            this.elevatorPosition = elevatorPosition;
+        }
+    }
+
+
+    static {
+        configureAngleMotors();
+        configureElevatorMotors();
+        configureAngleEncoder();
+        configureElevatorEncoder();
+    }
+
+    private static void configureAngleMotors() {
+        MASTER_ANGLE_MOTOR.restoreFactoryDefaults();
+        FOLLOWER_ANGLE_MOTOR.restoreFactoryDefaults();
+        MASTER_ANGLE_MOTOR.setInverted(MASTER_ANGLE_MOTOR_INVERTED);
+        FOLLOWER_ANGLE_MOTOR.setInverted(FOLLOWER_ANGLE_MOTOR_INVERTED);
+        MASTER_ANGLE_MOTOR.setIdleMode(ANGLE_MOTORS_IDLE_MODE);
+        FOLLOWER_ANGLE_MOTOR.setIdleMode(ANGLE_MOTORS_IDLE_MODE);
+        MASTER_ANGLE_MOTOR.enableVoltageCompensation(VOLTAGE_COMPENSATION_SATURATION);
+        FOLLOWER_ANGLE_MOTOR.enableVoltageCompensation(VOLTAGE_COMPENSATION_SATURATION);
+    }
+
+    private static void configureElevatorMotors() {
+        MASTER_ELEVATOR_MOTOR.restoreFactoryDefaults();
+        FOLLOWER_ELEVATOR_MOTOR.restoreFactoryDefaults();
+        MASTER_ELEVATOR_MOTOR.setInverted(MASTER_ELEVATOR_MOTOR_INVERTED);
+        FOLLOWER_ELEVATOR_MOTOR.setInverted(FOLLOWER_ELEVATOR_MOTOR_INVERTED);
+        MASTER_ELEVATOR_MOTOR.setIdleMode(ELEVATOR_MOTORS_IDLE_MODE);
+        FOLLOWER_ELEVATOR_MOTOR.setIdleMode(ELEVATOR_MOTORS_IDLE_MODE);
+        MASTER_ELEVATOR_MOTOR.enableVoltageCompensation(VOLTAGE_COMPENSATION_SATURATION);
+        FOLLOWER_ELEVATOR_MOTOR.enableVoltageCompensation(VOLTAGE_COMPENSATION_SATURATION);
+    }
+
+    private static void configureAngleEncoder() {
+        CANcoderConfiguration config = new CANcoderConfiguration();
+        config.MagnetSensor.MagnetOffset = ANGLE_MOTOR_OFFSET;
+        config.MagnetSensor.SensorDirection = ANGLE_MOTOR_DIRECTION;
+        config.MagnetSensor.AbsoluteSensorRange = ANGLE_MOTOR_RANGE;
+        ANGLE_ENCODER.getConfigurator().apply(config);
+    }
+
+    private static void configureElevatorEncoder() {
+
```

Comment:
Now, start working on this encoder. You won't need to configure the range since it's an elevator encoder. 
Configuring the offset will be a bit less intuitive. You need to set the current position of the encoder to the position it thinks it's in minus the offset. 
And direction is just a phase.